From 30b0eb0ac4cd3648b9d978e1fc186126fdf3fbdb Mon Sep 17 00:00:00 2001
From: Lu Wei <luwei32@huawei.com>
Date: Fri, 18 Nov 2022 16:42:18 +0800
Subject: [PATCH 0078/1398] [Huawei] Revert "[Backport] ipvlan: Modify the
 value of ipvlan modes"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I52H2U

--------------------------------

This reverts commit ca3c93d8b1d5d46453d21ed10837f625122c0934.

In order to make iproute compatible with other operation system like SUSE,
rethad and etc, put IPVLAN_MODE_L2E last.

Signed-off-by: Lu Wei <luwei32@huawei.com>
---
 drivers/net/ipvlan/ipvlan_main.c | 2 +-
 include/uapi/linux/if_link.h     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ipvlan/ipvlan_main.c b/drivers/net/ipvlan/ipvlan_main.c
index 4e0fa42c18d0..779a84f70914 100644
--- a/drivers/net/ipvlan/ipvlan_main.c
+++ b/drivers/net/ipvlan/ipvlan_main.c
@@ -14,7 +14,7 @@ int sysctl_ipvlan_loop_qlen = 131072;
 int sysctl_ipvlan_loop_delay = 10;
 static int ipvlan_default_mode = IPVLAN_MODE_L3;
 module_param(ipvlan_default_mode, int, 0400);
-MODULE_PARM_DESC(ipvlan_default_mode, "set ipvlan default mode: 0 for l2, 1 for l3, 2 for l2e, 3 for l3s, others invalid now");
+MODULE_PARM_DESC(ipvlan_default_mode, "set ipvlan default mode: 0 for l2, 1 for l3, 2 for l3s, 3 for l2e, others invalid now");
 
 static struct ctl_table_header *ipvlan_table_hrd;
 static struct ctl_table ipvlan_table[] = {
diff --git a/include/uapi/linux/if_link.h b/include/uapi/linux/if_link.h
index 0507ecc7275a..50d4705e1cbc 100644
--- a/include/uapi/linux/if_link.h
+++ b/include/uapi/linux/if_link.h
@@ -689,8 +689,8 @@ enum {
 enum ipvlan_mode {
 	IPVLAN_MODE_L2 = 0,
 	IPVLAN_MODE_L3,
-	IPVLAN_MODE_L2E,
 	IPVLAN_MODE_L3S,
+	IPVLAN_MODE_L2E,
 	IPVLAN_MODE_MAX
 };
 
-- 
2.17.1

